ykrt/src/mt.rs

```diff
@@ -27,7 +27,7 @@ use crate::print_jit_state;
 use crate::{
     compile::{default_compiler, CompiledTrace, Compiler, GuardId},
     location::{HotLocation, HotLocationKind, Location, TraceFailed},
-    trace::{default_tracer, RawTrace, ThreadTracer, Tracer},
+    trace::{default_tracer, RawTrace, TraceCollector, Tracer},
```

Comment:
Do you think that it's obvious that a `TraceCollector` is thread-specific?

I could buy that. Maybe we could clarify that in a docstring?

Reply:
I guess we could document that but I can't imagine what a non-thread-specific tracer would even look like. It feels a bit like overkill to me in the sense of "what other bonkers things should I tell you are bonkers"? But maybe I'm missing out on something.

Reply:
It's not bonkers though, is it?

Linux perf has a mode to trace all processes and threads on a whole CPU core:

> In per-cpu mode all processes (or processes from the selected cgroup i.e. -G option, or processes selected with -p or -u) are traced. Each cpu has its own buffer. Inheritance is allowed.

Reply:
Actually I was a good boy and the doc string is still:

```
/// Represents a thread which is currently tracing.
```

so done without a new commit!

Reply:
OK, good enough for me.